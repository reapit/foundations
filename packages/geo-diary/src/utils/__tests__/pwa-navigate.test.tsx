import React from 'react'
import { shallow } from 'enzyme'
import {
  PwaMessageHandler,
  PwaMessagePoster,
  PwaNavigateContextProps,
  PwaNavigateProvider,
  usePwaNavigate,
} from '../pwa-navigate'
import { renderHook, act } from '@testing-library/react-hooks'

describe('PwaNavigateProvider', () => {
  it('should match snapshot', () => {
    expect(shallow(<PwaNavigateProvider />)).toMatchSnapshot()
  })
})

describe('usePwaNavigate', () => {
  it('should return pwaNavState and setPwaNavState', async () => {
    const { result } = renderHook<{}, PwaNavigateContextProps>(() => usePwaNavigate(), {
      wrapper: (props) => <PwaNavigateProvider>{props.children}</PwaNavigateProvider>,
    })
    const mockUrl = 'http://some-url.com'

    expect(result.current.pwaNavState).toBe(null)

    act(() => {
      result.current.setPwaNavState(mockUrl)
    })

    expect(result.current.pwaNavState).toBe(mockUrl)
  })
})

describe('PwaMessageHandler', () => {
  it('should match snapshot', () => {
    expect(shallow(<PwaMessageHandler setPwaNavState={jest.fn} pwaNavState="http://some-url" />)).toMatchSnapshot()
  })
})

describe('PwaMessagePoster', () => {
  it('should match snapshot', () => {
    expect(shallow(<PwaMessagePoster />)).toMatchSnapshot()
  })
})
