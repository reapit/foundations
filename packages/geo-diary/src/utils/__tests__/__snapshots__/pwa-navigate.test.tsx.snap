// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PwaMessageHandler should match snapshot 1`] = `""`;

exports[`PwaMessagePoster should match snapshot 1`] = `""`;

exports[`PwaNavigateProvider should match snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "pwaNavState": null,
      "setPwaNavState": [Function],
    }
  }
/>
`;
